src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/ProvideObsoleteAttributeMessage.cs

```diff
@@ -75,8 +75,11 @@ private static void AnalyzeSymbol(SymbolAnalysisContext context, INamedTypeSymbo
                     if (attribute.ConstructorArguments.IsEmpty ||
                         string.IsNullOrEmpty(attribute.ConstructorArguments.First().Value as string))
                     {
-                        SyntaxNode node = attribute.ApplicationSyntaxReference.GetSyntax(context.CancellationToken);
-                        context.ReportDiagnostic(node.CreateDiagnostic(Rule, context.Symbol.Name));
+                        SyntaxNode? node = attribute.ApplicationSyntaxReference?.GetSyntax(context.CancellationToken);
+                        if (node != null)
```

Comment:
This doesn't seem correct. We should not bail out on reporting diagnostics based on lack of syntax. Couple of questions:

1. Do we know if null is possible? As in, does `ISymbol.GetAttributes` return attributes inherited from base types, which can be in metadata? I thought it only returns attributes applied directly to the symbol, and given the analyzer callbacks only happen for source symbols, all attributes should be in source. If so, null value is not possible.
2. If null is possible, can we find an alternate syntax to report diagnostic (say symbol localtion), instead of bailing out?

Reply:
There is not much doc available for `GetAttributes` so that's pretty hard to say but I would say that if this is true then I don't get the issue described in the ticket.

Reply:
@Evangelink The issue specifically identified an NRE in DoNotMixAttributesFromDifferentVersionsOfMEFAnalyzer, which seems to walk the base type chain to get all applicable attributes - so for that analyzer the issue is certainly possible. We should probably narrow down this PR to fix just the cases that use GetApplicableAttributes? Also the fix should not be to bail out, but instead to report on an alternate, reasonable syntax.